### DISCUSSION TITLE
Issue with M3 Command when using the same tool for 2 Operations Consecutively

### DISCUSSION THREAD

**Discussion by 3dPrintingMD** [TimPaterson/Fusion360-Batch-Post#26] [General]
Just want to start off with a thank you, this is working very good so far.

One thing I noticed is if you have 2 operations in a row using the same tool, it will not recognize this, and will command an M3.

Example:

(2D Chamfer1)
N460 G28 G91 Z0.
N465 G90
N470 M1
N475 T6 M6
N480 S18000 M3
N485 G54
N490 M7
N495 G0 X1.905 Y0.
N500 G43 Z15. H6
N505 Z5.
N510 G1 Z2. F351.
N515 Z-1.635
N520 X2.54 F2140.
N525 G3 X-2.54 I-2.54 J0.
N530 X2.54 I2.54 J0.
N535 G1 X1.905
N540 G0 Z15.
N545 M9
(2D Chamfer2)
**N550 S18000 M3
N555 G54
N560 M7**
N565 G0 X0. Y-11.27
N570 G43 Z15. H6
N575 Z5.
N580 G1 Z2. F351.
N585 Z-0.635
N590 Y-10.635 F2140.
N595 X-10.
N600 G2 X-10.635 Y-10. I0. J0.635
N605 G1 Y10.
N610 G2 X-10. Y10.635 I0.635 J0.
N615 G1 X10.
N620 G2 X10.635 Y10. I0. J-0.635
N625 G1 Y-10.
N630 G2 X10. Y-10.635 I-0.635 J0.
N635 G1 X0.
N640 Y-11.27
N645 G0 Z15.
N650 M9
N655 G28 G91 Z0.
N660 G90
N665 G28 G91 X0. Y0.
N670 G90
N675 M30
%

Edited to clarify

**Comment by TimPaterson** [comment]
The G-code for a tool change is Txx.  While pasting the operations together, PostProcessAll tracks the current tool number and deletes the line with a tool change if it is the same as the current tool.  So line N475 has T6 M6 followed by a line for spindle speed, but on the second operation there is no Txx command before the spindle speed on line N550.

M3 is start spindle forward, and it is normal for it to be on the same line as set spindle speed, Sxx.  I don't see an issue here.

**Reply by 3dPrintingMD** [reply]
Thank you for the reply. My M3 command has a Spindle Delay in it, so every time it comes up the machine will pause(delay). Not really an issue though.
